jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -207,7 +228,36 @@ private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
 				bufferPrinter.println();
 			}
 			bufferPrinter.flush();
-			result = DiagnosticProperties.makeStringResult(buffer.toString());
+			result = makeStringResult(buffer.toString());
+		}
+		return result;
+	}
+
+	private static DiagnosticProperties doDump(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		logMessage("doDump: ", diagnosticCommand); //$NON-NLS-1$
+		if (parts.length != 2) {
+			result =  makeErrorProperties("Error: wrong number of arguments"); //$NON-NLS-1$
+		} else {
+			String pathString = parts[1];
+			String dumpType;
+			/* handle legacy form of Dump.heap for compatibility with reference implementation */
+			if (DIAGNOSTICS_GC_HEAP_DUMP.equals(parts[0])) {
+				dumpType = "heap"; //$NON-NLS-1$
+			} else {
+				dumpType = parts[0].split("\\.")[1]; //$NON-NLS-1$
+			}
+			String fileDirective = ("system".equals(dumpType) &&  IPC.isZOS) ? ":dsn=" : ":file="; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+
+			String request = dumpType + fileDirective + pathString;
+			try {
+				String actualDumpFile = triggerDumpsImpl(request, dumpType + "DumpToFile"); //$NON-NLS-1$
+				result = makeStringResult("Dump written to " + actualDumpFile); //$NON-NLS-1$
+			} catch (InvalidDumpOptionExceptionBase e) {
+				logMessage("doDump exception: ", e.getMessage()); //$NON-NLS-1$
+				DiagnosticProperties.makeExceptionProperties(e);
```

Comment:
Any particularly reason `DiagnosticProperties.makeExceptionProperties()` isn't imported statically like `DiagnosticProperties.makeErrorProperties()` or `DiagnosticProperties.makeStringResult`?

Reply:
My thought was about the consistency of `DiagnosticProperties` usages, but would like to listen why `DiagnosticProperties.makeExceptionProperties()` is different from those two.

Reply:
> Any particularly reason `DiagnosticProperties.makeExceptionProperties()` isn't imported statically like `DiagnosticProperties.makeErrorProperties()` or `DiagnosticProperties.makeStringResult`?

I generally `import static` only items which are used multiple times.

Reply:
I suggest `import static` should only be used when it doesn't reduce the readability of the code. IMO, none of the uses in the pull request qualify. An unqualified reference is more work for the (human) reader to understand.

Reply:
Thanks for the suggestion, Keith.  I will keep that in mind.